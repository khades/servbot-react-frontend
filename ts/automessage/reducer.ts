import States from "../utils/states";
import { actiontypes, AutoMessageAction } from "./actions";
import { IAutoMessageWithHistory } from "./types";

export interface IAutoMessageState {
    channelID: string;
    id: string;
    isNew: boolean;
    state: States;
    validationError: boolean;
    content?: IAutoMessageWithHistory;
}

const initialState: IAutoMessageState = {
    channelID: "",
    id: "",
    isNew: false,
    state: States.NOTINITIATED,
    validationError: false,
};

const reducer = (state: IAutoMessageState = initialState, action: AutoMessageAction): IAutoMessageState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.LOADING,
            });
        case actiontypes.CREATENEW:
            return {
                channelID: action.payload.channelID,
                content: {
                    channelID: action.payload.channelID,
                    durationLimit: 300000000000,
                    durationThreshold: 0,
                    game: "",
                    history: [],
                    id: "",
                    message: "",
                    messageLimit: 50,
                    messageThreshold: 0,
                },
                id: "",
                isNew: true,
                state: States.READY,
                validationError: false,
            };
        case actiontypes.READY:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                content: action.payload.content,
                id: action.payload.id,
                isNew: false,
                state: States.READY,
                validationError: false,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        case actiontypes.SAVE:
            return Object.assign({}, state, {
                state: States.UPDATING,
            });
        case actiontypes.SAVENEW:
            return Object.assign({}, state, {
                state: States.UPDATING,
            });
        case actiontypes.ONSAVEERROR:
            return Object.assign({}, state, {
                state: States.READY,
                validationError: true,
            });
        case actiontypes.RESET:
            return initialState;
        case actiontypes.AFTERCREATION:
            return Object.assign({}, state, { id: action.payload.id });
        default:
            return state;
    }
};

export default reducer;
