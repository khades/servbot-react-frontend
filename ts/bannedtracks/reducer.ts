import States from "../utils/states";
import { actiontypes, BannedTracksAction } from "./actions";
import { IBannedTracks } from "./types";

export interface IBannedTracksState {
    state: States;
    page: number;
    content?: IBannedTracks;
}

const initialState: IBannedTracksState = {
    page: 1,
    state: States.NOTINITIATED,
};

const reducer = (state: IBannedTracksState = initialState, action: BannedTracksAction): IBannedTracksState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return {
                page: 1,
                state: States.LOADING,
            };
        case actiontypes.READY:
            return Object.assign({}, state, {
                content: action.payload.content,
                page: action.payload.page,
                state: States.READY,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        case actiontypes.UNBAN:
            return Object.assign({}, state, {
                state: States.UPDATING,
            });
        case actiontypes.RESET:
            return initialState;
        default:
            return state;
    }
};

export default reducer;
