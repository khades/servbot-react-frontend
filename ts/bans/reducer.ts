import { IBan } from "../api/types";
import States from "../utils/states";
import { actiontypes, BansAction } from "./actions";

export interface IBansState {
    bans?: IBan[];
    state: States;
    channelID: string;
}

const initialState = {
    channelID: "",
    state: States.NOTINITIATED,
};

const reducer = (state: IBansState = initialState, action: BansAction): IBansState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, action.payload, { state: States.LOADING });
        case actiontypes.READY:
            return Object.assign({}, action.payload, { state: States.READY });
        case actiontypes.NOTFOUND:
            return Object.assign({}, action.payload, { state: States.NOTFOUND });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, action.payload, { state: States.FORBIDDEN });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, action.payload, { state: States.NOTAUTHORIZED });
        case actiontypes.OFFLINE:
            return Object.assign({}, action.payload, { state: States.OFFLINE });
        default:
            return state;
    }
};

export default reducer;
