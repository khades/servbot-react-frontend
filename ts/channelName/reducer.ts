import States from "../utils/states";
import actions, { actiontypes } from "./actions";

export interface IChannelName {
    name?: string;
    state: States;
}
export interface IChannelNameStore {
    [key: string]: IChannelName;
}

const reducer = (state: IChannelNameStore = {}, action: actions): IChannelNameStore => {
    switch (action.type) {
        case actiontypes.READY:
            return Object.assign(state, {}, {
                [action.payload.channelNameID]: {
                    name: action.payload.channelName,
                    state: States.READY,
                },
            });
        case actiontypes.NOTFOUND:
            return Object.assign(state, {}, {
                [action.payload.channelNameID]: {
                    state: States.NOTFOUND,
                },
            });
        case actiontypes.LOADING:
            return Object.assign(state, {}, {
                [action.payload.channelNameID]: {
                    state: States.LOADING,
                },
            });
        default:
            return state;
    }
};

export default reducer;
