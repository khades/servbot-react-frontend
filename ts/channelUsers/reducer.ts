import { IUserLogsInfo } from "../userLogs/types";
import States from "../utils/states";
import { actiontypes, ChannelUsersAction } from "./actions";

export interface IChannelUsersState {
    channelID: string;
    userName: string;
    users: IUserLogsInfo[];
    state: States;
}

const initialState: IChannelUsersState = {
    channelID: "",
    state: States.NOTINITIATED,
    userName: "",
    users: [],
};

const reducer = (state: IChannelUsersState = initialState, action: ChannelUsersAction) => {

    switch (action.type) {
        case actiontypes.READY:
            return Object.assign({}, state, {
                state: States.READY,
                users: action.payload,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, { state: States.OFFLINE });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.FORBIDDEN,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, { state: States.NOTAUTHORIZED });
        case actiontypes.LOADING:
            if (state.channelID === action.payload.channelID) {
                return Object.assign({}, state, {
                    channelID: action.payload.channelID,
                    state: States.UPDATING,
                    userName: action.payload.userName,
                });
            } else {
                return {
                    channelID: action.payload.channelID,
                    state: States.LOADING,
                    userName: action.payload.userName,
                    users: [],
                };
            }
        case actiontypes.RESET:
            return initialState;
        default:
            return state;
    }
};

export default reducer;
