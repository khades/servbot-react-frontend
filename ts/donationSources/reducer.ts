import States from "../utils/states";
import { actiontypes, DonationSourcesAction } from "./actions";
import { IDonationSources } from "./types";

export interface IDonationSourcesState {
    channelID: string;
    state: States;
    content?: IDonationSources;
}

const initialState: IDonationSourcesState = {
    channelID: "",
    state: States.NOTINITIATED,
};

const reducer = (state: IDonationSourcesState = initialState, action: DonationSourcesAction): IDonationSourcesState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.LOADING,
            });
        case actiontypes.READY:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                content: action.payload.content,
                state: States.READY,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        default:
            return state;
    }
};

export default reducer;
