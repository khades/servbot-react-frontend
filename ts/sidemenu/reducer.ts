import { actiontypes, SideMenuActions } from "./actions";

export enum SideMenuStates {
    INIT = "INIT",
    HIDDEN = "HIDDEN",
    SHOWN = "SHOWN",
}

export interface ISideMenuStore {
    state: SideMenuStates;
}

const reducer = (state: ISideMenuStore = { state: SideMenuStates.INIT }, action: SideMenuActions): ISideMenuStore => {
    switch (action.type) {
        case actiontypes.SHOW:
            return { state: SideMenuStates.SHOWN };
        case actiontypes.HIDE:
            return { state: SideMenuStates.HIDDEN };
        default:
            return state;
    }
};

export default reducer;
