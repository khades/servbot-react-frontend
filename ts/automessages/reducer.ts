import { IAutoMessageWithHistory } from "../automessage/types";
import States from "../utils/states";
import { actiontypes, AutoMessagesAction } from "./actions";

export interface IAutoMessagesState {
    channelID: string;
    state: States;
    emptyShown: boolean;
    content?: IAutoMessageWithHistory[];
}

const initialState: IAutoMessagesState = {
    channelID: "",
    emptyShown: false,
    state: States.NOTINITIATED,
};

const reducer = (state: IAutoMessagesState = initialState, action: AutoMessagesAction): IAutoMessagesState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.LOADING,
            });
        case actiontypes.READY:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                content: action.payload.content,
                emptyShown: false,
                state: States.READY,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        case actiontypes.SHOWEMPTY:
            return Object.assign({}, state, {
                emptyShown: true,
            });
        case actiontypes.HIDEEMPTY:
            return Object.assign({}, state, {
                emptyShown: false,
            });
        default:
            return state;
    }
};

export default reducer;
