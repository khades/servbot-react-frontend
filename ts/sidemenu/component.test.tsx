import * as React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { create } from "react-test-renderer";
import configureStore from "redux-mock-store";
import { IChannelNameStore } from "../channelName/reducer";
import * as Routes from "../routes/routes";
import States from "../utils/states";
import SideMenu, { ISideMenuProps } from "./component";
import { SideMenuStates } from "./reducer";
describe("Side Menu", () => {
    const storeState: IChannelNameStore = {
        2342342: {
            name: "User",
            state: States.READY,
        },
        2345: {
            name: "Channel",
            state: States.READY,
        },
    };
    const store = configureStore()({ channelName: storeState });
    it("should render properly for mods", () => {
        const props: ISideMenuProps = {
            hideMenu: () => { },
            isModOnChannel: true,
            menuState: SideMenuStates.SHOWN,
            userInfo: {
                avatarUrl: "",
                currentChannel: "2345",
                modChannels: [],
                state: States.READY,
                userID: "2342342",
                username: "63636363",
            },
        };
        const notifications = create(
            <Provider store={store}>
                <MemoryRouter initialEntries={[Routes.ToBans(props.userInfo.currentChannel)]}>
                    <SideMenu {...props} />
                </MemoryRouter>
            </Provider>,
        );
        expect(notifications.toJSON()).toMatchSnapshot();
    });

    it("should render properly for non-mods", () => {
        const props: ISideMenuProps = {
            hideMenu: () => { },
            isModOnChannel: false,
            menuState: SideMenuStates.SHOWN,
            userInfo: {
                avatarUrl: "",
                currentChannel: "2345",
                modChannels: [],
                state: States.READY,
                userID: "2342342",
                username: "63636363",
            },
        };
        const notifications = create(
            <Provider store={store}>

                <MemoryRouter initialEntries={[Routes.ToTemplates(props.userInfo.currentChannel)]}>
                    <SideMenu {...props} />
                </MemoryRouter>
            </Provider>,
        );
        expect(notifications.toJSON()).toMatchSnapshot();
    });
    it("should not render if not ready", () => {
        const props: ISideMenuProps = {
            hideMenu: () => { },
            isModOnChannel: false,
            menuState: SideMenuStates.SHOWN,
            userInfo: {
                avatarUrl: "",
                currentChannel: "",
                modChannels: [],
                state: States.LOADING,
                userID: "",
                username: "",
            },
        };
        const notifications = create(
            <Provider store={store}>
                <MemoryRouter initialEntries={[Routes.ToTemplates("1231")]}>
                    <SideMenu {...props} />
                </MemoryRouter>
            </Provider> ,
        );
        expect(notifications.toJSON()).toMatchSnapshot();
    });
});
