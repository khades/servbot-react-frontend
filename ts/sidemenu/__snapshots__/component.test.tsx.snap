// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Menu should not render if not ready 1`] = `
<div
  className="site-menu site-menu--shown"
>
  <div
    className="site-menu__container"
    onClick={[Function]}
  />
</div>
`;

exports[`Side Menu should render properly for mods 1`] = `
<div
  className="site-menu site-menu--shown"
>
  <div
    className="site-menu__container"
    onClick={[Function]}
  >
    <a
      className="site-menu__header"
      href="/channel/2345"
      onClick={[Function]}
    >
      <img
        src=""
      />
      <div
        className="site-menu__header-info"
      >
        <div>
          63636363
           
        </div>
        <span>
          Channel 
          <span
            className="channel-name"
          >
            Channel
          </span>
        </span>
      </div>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/logs"
      onClick={[Function]}
    >
      <span>
        Message logs
      </span>
    </a>
    <a
      aria-current="page"
      className="site-menu__link site-menu__link--selected"
      href="/channel/2345/bans"
      onClick={[Function]}
      style={Object {}}
    >
      <span>
        Bans
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/templates"
      onClick={[Function]}
    >
      <span>
        Chat Commands
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/subAlert"
      onClick={[Function]}
    >
      <span>
        Subscription alerts
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/autoMessages"
      onClick={[Function]}
    >
      <span>
        Automessages
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/songrequests"
      onClick={[Function]}
    >
      <span>
        Song requests
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/subdays"
      onClick={[Function]}
    >
      <span>
        Subdays
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/subs"
      onClick={[Function]}
    >
      <span>
        Subscriptions
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/externalservices"
      onClick={[Function]}
    >
      <span>
        External Services
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/subtrain"
      onClick={[Function]}
    >
      <span>
        Subtrain
      </span>
    </a>
  </div>
</div>
`;

exports[`Side Menu should render properly for non-mods 1`] = `
<div
  className="site-menu site-menu--shown"
>
  <div
    className="site-menu__container"
    onClick={[Function]}
  >
    <a
      className="site-menu__header"
      href="/channel/2345"
      onClick={[Function]}
    >
      <img
        src=""
      />
      <div
        className="site-menu__header-info"
      >
        <div>
          63636363
           
        </div>
        <span>
          Channel 
          <span
            className="channel-name"
          >
            Channel
          </span>
        </span>
      </div>
    </a>
    <a
      aria-current="page"
      className="site-menu__link site-menu__link--selected"
      href="/channel/2345/templates"
      onClick={[Function]}
      style={Object {}}
    >
      <span>
        Chat Commands
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/songrequests"
      onClick={[Function]}
    >
      <span>
        Song requests
      </span>
    </a>
    <a
      aria-current={null}
      className="site-menu__link"
      href="/channel/2345/subdays"
      onClick={[Function]}
    >
      <span>
        Subdays
      </span>
    </a>
  </div>
</div>
`;
